internal/lexer/token.go

```diff
@@ -0,0 +1,51 @@
+package lexer
+
+type TokenType int
+
+const (
+	EOF TokenType = iota
+	IDENTITY
+	NUMBER
+	STRING
+
+	PATH     // .
```

Comment:
I think that treating the dot (.) as a token is too granular. AFAIU (please correct me if I"m wrong), the token should be something that deserves to be a node in the AST. So it the AST tree we should care whether we have a regular "PATH" and it value or a "CAPTURE_PATH".

Reply:
This is just the lexer, the parser is the one that will understand what is a Node and what's not.